src/input/json_to_storage.cc

```diff
@@ -568,11 +568,11 @@ StorageBase * JSONToStorage::make_storage(JSONPath &p, const Type::Bool *bool_ty
 StorageBase * JSONToStorage::make_storage(JSONPath &p, const Type::Integer *int_type)
 {
     if (p.head()->type() == json_spirit::int_type) {
-        boost::int64_t value = p.head()->get_int64();
+        std::int64_t value = p.head()->get_int64();
 
-        if ( value >= std::numeric_limits<int>::min() &&
-             value <= std::numeric_limits<int>::max() &&
-             int_type->match(value) )
+        if ( value >= -(std::int64_t)std::numeric_limits<unsigned int>::max() &&
```

Comment:
- remove this check
- change bounds of Integer to int64, as well as argument of match()